frontend/src/component/demo/DemoSteps/DemoSteps.tsx

```diff
@@ -59,54 +60,58 @@ export const DemoSteps = ({
     topic,
     setTopic,
     topics,
+    onFinish,
 }: IDemoStepsProps) => {
     const theme = useTheme();
     const navigate = useNavigate();
     const location = useLocation();
     const [run, setRun] = useState(false);
-    const [flow, setFlow] = useState<'next' | 'back'>('next');
+    const [flow, setFlow] = useState<'next' | 'back' | 'load'>('load');
 
     const abortController = new AbortController();
 
+    const setTopicStep = (topic: number, step?: number) => {
+        setRun(false);
+        setTopic(topic);
+        if (step !== undefined) {
+            setSteps(steps => {
+                const newSteps = [...steps];
+                newSteps[topic] = step;
+                return newSteps;
+            });
+        }
+    };
+
     const skip = () => {
         abortController.abort();
-        setTopic(-1);
+        setTopicStep(-1);
         setExpanded(false);
     };
 
-    const setStep = (topic: number, step: number) => {
-        setSteps(steps => {
-            const newSteps = [...steps];
-            newSteps[topic] = step;
-            return newSteps;
-        });
-    };
-
     const back = () => {
         setFlow('back');
         if (steps[topic] === 0) {
             const newTopic = topic - 1;
-            setTopic(newTopic);
-            setStep(newTopic, topics[newTopic].steps.length - 1);
+            setTopicStep(newTopic, topics[newTopic].steps.length - 1);
         } else {
-            setStep(topic, steps[topic] - 1);
+            setTopicStep(topic, steps[topic] - 1);
```

Comment:
Does it make sense to protect from potentially a `-2` here? Something like:
```suggestion
            setTopicStep(topic, min(steps[topic] - 1, -1));
```

Reply:
Maybe something to refactor and change the logic a bit in the future. Will not prioritize this for now but will keep it in mind.